### PR TITLE
Add docs: Type Support In Koalas

### PR DESCRIPTION
**Pull request by itholic**
Added documents "Type Support In Koalas" to describe which type is supported in Koalas and which type is not.

data_type_docs

### CODE REVIEW

File: docs/source/user_guide/types.rst [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+This chapter gives you an information of what types are supported and which types are not.
+
+
+Types supported in Koalas
+-------------------------
+
+A table below shows which NumPy types are matched to which PySpark types internally in Koalas.
+
+============= =======================
+NumPy         PySpark
```
Comment (HyukjinKwon) [inline docs/source/user_guide/types.rst:21]: I think we don't have to expose our internal mapping. PySpark is just an internal engine in Koalas.

File: docs/source/user_guide/types.rst [hunk 2]
```diff
@@ -0,0 +1,145 @@
+======================
+Type Support In Koalas
+======================
+
+.. currentmodule:: databricks.koalas
+
+
+This is based on Koalas 1.1.4.
+
+Koalas supports various of types by mapping them to specific types in PySpark internally.
```
Comment (HyukjinKwon) [inline docs/source/user_guide/types.rst:10]: I wouldn't focus on internal stuff in the user-facing docs. We should better just document user-facing things. Like, what types you get if you convert a Spark DataFrame to Koalas DataFrame.

Comment (HyukjinKwon) [reply docs/source/user_guide/types.rst:10]: Let's focus on pandas DataFrame <> Koalas DataFrame <> Spark DataFrame instead of documenting internal stuffs

Comment (itholic) [reply docs/source/user_guide/types.rst:10]: Sounds good. Okay I'll address it, thanks!

File: docs/source/user_guide/types.rst [hunk 3]
```diff
@@ -0,0 +1,145 @@
+======================
+Type Support In Koalas
+======================
+
+.. currentmodule:: databricks.koalas
+
+
+This is based on Koalas 1.1.4.
```
Comment (ueshin) [inline docs/source/user_guide/types.rst:8]: I don't think we need this.
If we change the type support, we should also update this doc.

File: docs/source/user_guide/types.rst [hunk 4]
```diff
@@ -0,0 +1,145 @@
...
+    StringType
+
+    >>> ks.Series([[False, True, False]]).spark.data_type
+    ArrayType(BooleanType,true)
+
+
+pandas types not supported in Koalas
+------------------------------------
+
+For reference, this is based on pandas 1.1.4.
```
Comment (ueshin) [inline docs/source/user_guide/types.rst:128]: ditto.

File: docs/source/user_guide/types.rst (hunk 3 above)
Comment (itholic) [reply docs/source/user_guide/types.rst:8]: Sounds reasonable. Thanks!!

File: docs/source/user_guide/types.rst [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+    decimal              object
+    float               float32
+    double              float64
+    integer               int32
+    long                  int64
+    short                 int16
+    timestamp    datetime64[ns]
+    string               object
+    boolean                bool
+    dtype: object
```
[the remark this answers is not in the document]
Comment (ueshin) [reply docs/source/user_guide/types.rst:75]: Actually I think the mapping tables in the previous commit:
- https://github.com/databricks/koalas/pull/1906/commits/46f69d9cc0aae6334594338c509bd102084b0df4#diff-42dd0ba227b3dfc8a15ce59f5cb687fcf08cc92b0c75cf34e2da8277984ee76fR20-R40
- https://github.com/databricks/koalas/pull/1906/commits/46f69d9cc0aae6334594338c509bd102084b0df4#diff-42dd0ba227b3dfc8a15ce59f5cb687fcf08cc92b0c75cf34e2da8277984ee76fR20-R40

or the table you made in our offline discussion are useful.

File: docs/source/user_guide/types.rst [hunk 6]
```diff
@@ -0,0 +1,103 @@
...
+
+We can easily convert Koalas DataFrame to pandas DataFrame, and the data types are basically same as pandas.
+
+.. code-block:: python
+
+    # Convert Koalas DataFrame to pandas DataFrame
+    >>> pdf = kdf.to_pandas()
+
+    # Check the pandas data types
+    >>> pdf.dtypes
```
Comment (xinrong-databricks) [inline docs/source/user_guide/types.rst:64]: The document looks pretty clear to me! nit: Within the `Type casting between pandas and Koalas` section itself, we may not tell that column names indicate respective PySpark types. I'm not sure whether we want to point this out, just FYI.

File: docs/source/user_guide/types.rst [hunk 7]
```diff
@@ -0,0 +1,103 @@
...
+When converting the Koalas DataFrame from/to PySpark DataFrame, the data types automatically casted to the appropriate type.
+
+The example below shows how types are casted between PySpark DataFrame and Koalas DataFrame.
+
+.. code-block:: python
+
+    # 1. Create PySpark DataFrame
+    >>> sdf = spark.createDataFrame([
+    ...     (1, Decimal(1.0), 1., 1., 1, 1, 1, datetime(2020, 10, 27), "1", True),
+    ... ], 'tinyint tinyint, decimal decimal, float float, double double, integer integer, long long, short short, timestamp timestamp, string string, boolean boolean')
```
[the remark this answers is not in the document]
Comment (itholic) [reply docs/source/user_guide/types.rst:22]: Oh, thanks for the catch!

File: docs/source/user_guide/types.rst (hunk 5 above)
[the remark this answers is not in the document]
Comment (itholic) [reply docs/source/user_guide/types.rst:75]: @ueshin Thanks for the opinion, I'll add the table to somewhere in the docs!

File: docs/source/user_guide/types.rst (hunk 6 above)
Comment (itholic) [reply docs/source/user_guide/types.rst:64]: Thanks for the opinion, @xinrong-databricks ! Let me check about this :)

**Pull request merged**